File: GithubSearch(OC)/GithubSearch/View/RepoTableViewCell.m
//
//  RepoTableViewCell.m
//  GithubSearch
//
//  Created by Nathan on 20/02/2018.
//  Copyright © 2018 Nathan. All rights reserved.
//

#import "RepoTableViewCell.h"
#import <SDWebImage/UIImageView+WebCache.h>

@implementation RepoTableViewCell
- (void) updateUI {
    [self.nameLabel setText:self.repoInfo.name];
    [self.descriptionLabel setText:self.repoInfo.description];
    [self.repoImage sd_setImageWithURL: self.repoInfo.image_url placeholderImage:nil];
}
@end

File: GithubSearchDemo-finished/GithubSearchDemo/View/RepoTableViewCell.swift
//
//  RepoTableViewCell.swift
//  GithubSearchDemo
//
//  Created by Nathan on 05/12/2017.
//  Copyright © 2017 Nathan. All rights reserved.
//

import UIKit
import SDWebImage

class RepoTableViewCell: UITableViewCell {
    
    @IBOutlet weak var repoImage: UIImageView!
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var descriptionLabel: UILabel!
    
    var repoInfo: RepositoryInfo? {
        didSet {
            updateUI()
        }
    }
    
    private func updateUI() {
        self.nameLabel.text = repoInfo?.name
        self.descriptionLabel.text = repoInfo?.description
        self.repoImage.sd_setImage(with: repoInfo?.image_url)
    }
}
